ArduCopter/mode_acro.cpp

```diff
@@ -10,21 +10,34 @@
 
 void Copter::ModeAcro::run()
 {
+    // convert the input to the desired body frame rate
     float target_roll, target_pitch, target_yaw;
+    get_pilot_desired_angle_rates(channel_roll->get_control_in(), channel_pitch->get_control_in(), channel_yaw->get_control_in(), target_roll, target_pitch, target_yaw);
 
-    // if not armed set throttle to zero and exit immediately
-    if (!motors->armed() || ap.throttle_zero || !motors->get_interlock()) {
-        zero_throttle_and_relax_ac();
-        return;
+    if (!motors->armed()) {
```

Comment:
this block is use numerous time, we could fatorize it

Reply:
yes, i agree but I'll leave that out-of-scope for the moment if that's OK because it's an existing issue in master and this PR is already scary enough :-)